plugins/inventory/vmware_vm_inventory.py

```diff
@@ -68,19 +69,69 @@
             description:
             - Specify the list of VMware schema properties associated with the VM.
             - These properties will be populated in hostvars of the given VM.
-            - Each value in the list specifies the path to a specific property in VM object.
+            - Each value in the list can be a path to a specific property in VM object or a path to a collection of VM objects.
+            - In addition to VM properties, the following are special values
+            - Use 'customValue' to populate VM's custom attributes.
+            - Use 'all' to populate all VM properties.
             type: list
             default: [ 'name', 'config.cpuHotAddEnabled', 'config.cpuHotRemoveEnabled',
                        'config.instanceUuid', 'config.hardware.numCPU', 'config.template',
-                       'config.name', 'guest.hostName', 'guest.ipAddress',
+                       'config.name', 'config.uuid' ,'guest.hostName', 'guest.ipAddress',
                        'guest.guestId', 'guest.guestState', 'runtime.maxMemoryUsage',
-                       'customValue'
+                       'customValue', 'summary.runtime.powerState', 'config.guestId'
                        ]
+            version_added: "2.9"
+        resources:
+            description:
+            - A list of resources to limit search scope.
+            - Each resource item is represented by exactly one C('vim_type_snake_case):C(list of resource names) pair and optional nested I(resources)
```

Comment:
This is a mouthful. It leaves so questions open in terms of if the side-by-side entries are ANDed or ORed together, implicitly the listed values must be ORed, and if the nested properties are ANDed or ORed with the whole rest of the query. It only does equal / not-equal.

The old script had an option that looked like this:

```
host_filters={{ runtime.powerstate == "poweredOn" }}
```

There's no way to convert that to this programatically. I want to ask if this is intended to be "the" solution for host filtering.

The name "resources" also doesn't naturally make me think of host filtering.

Reply:
No, it is different from `host_filters`. `host_filters` filter VMs based on its properties and it's a post-process(client-side). It will be in a new PR(see previous comments for more details). 


> ```yaml
> host_filters:
> - runtime.powerstate == "poweredOn"
> - runtime.maxmemoryusage >= 512
> ```

`resources` allow limiting the scope of retrieving VMs from host, based on resources(datacenter, clusters, folder, pool, ...etc) that it is belong to. `resources` can be nested to resemble tree/directory structure which is native to vCenter Web UI. Since the process is done on the server-side, it improves performance specially for big deployment.

 The following is an example to target VMs in `dev` folders in `Asia-Datacenter1` and  `Asia-Datacenter3` and `prod` folder in `Europe-Datacenter5`:
 
 ```yaml
 resources:
 - datacenter:
   - Asia-Datacenter1
   - Asia-Datacenter3
   resources:
   - folder:
     - dev
 - datacenter:
   - Europe-Datacenter5
   resources:
   - folder:
     - prod
 ```